### DISCUSSION TITLE
How to install poetry in a DevContainer environment? -- I am confused about access rights.

### DISCUSSION THREAD

**Discussion by PeterBosch2** [python-poetry/poetry#8182] [Q&A]
I spent quite some time in getting `poetry` properly installed into a DevContainer. Access rights are the pain or the question whether to install `poetry` as root or as a container user, such as "vscode". 

- I understand `pipx` as the suggested method to install poetry in an environment. So, issuing `pipx` in the Dockerfile (as root) seems to be the way to go.  
  ```dockerfile
  RUN pipx install poetry==${POETRY_VERSION} --python="$(which python)" \
     && poetry completions bash > /etc/bash_completion.d/poetry
  ```
  However, this approach leads to a dead end when the container user wants to install something to `poetry self`
  ```bash
  poetry self add "poetry-dynamic-versioning[plugin]"
  ```
  this fails for obvious reasons as `poetry` wants to install to the `venv` created by `pipx` as root. 

- Installing poetry as the container user is a bit of guess work "who the container user will be". Changes to the container image might lead to a different user.

What is the best way to go? Any suggestions/hints/help are appreciated!

**Comment by clintonroy** [comment]
I still don't actually know what a dev container is, so this might not be useful, I'm very confused as to how they're supposed to be used. e.g. installing tab completions in a container seems useless to me. Can you not install poetry as the user that's going to run the container?

USER user
pipx install poetry dadada

?

**Reply by PeterBosch2** [reply]
https://github.com/devcontainers
